lms/lmsweb/views.py

```diff
@@ -343,41 +363,55 @@ def upload_page():
     })
 
 
-@webapp.route(f'{routes.DOWNLOADS}/<int:solution_id>')
-@webapp.route(f'{routes.DOWNLOADS}/<int:solution_id>/<int:file_id>')
+@webapp.route(f'{routes.DOWNLOADS}/<string:download_id>')
 @login_required
-def download(solution_id: int, file_id: Optional[int] = None):
-    solution = Solution.get_or_none(solution_id)
-    if solution is None:
+def download(download_id: str):
+    """Downloading a zip file of the code files.
+
+    Args:
+        download_id (str): Can be a solution.id or sharedsolution.shared_url.
```

Comment:
Comments have to be max 72 chars long.

Split to two lines, add ` on each side of `solution.id` and `sharedsolution.shared_url`.